### PR TITLE
feat: Add DiscountCondition entity + Join table per relation

### PR DESCRIPTION
**Pull request by olivermrbl**
Solves #1143 

Add entities for discount conditions. See [this discussion](https://github.com/medusajs/medusa/discussions/1148) for details on and reasoning behind the implementation.

### CODE REVIEW

File: packages/medusa/src/models/discount-rule-condition.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+  @DbAwareColumn({
+    type: "enum",
+    enum: DiscountRuleConditionType,
+  })
+  type: DiscountRuleConditionType
+
+  @Index()
+  @Column()
+  rule_id: string
```
Comment (srindom) [inline packages/medusa/src/models/discount-rule-condition.ts:47]: Should either be `discount_rule_id` or the join key name below should be changed to `rule_id`

**Review by srindom — CHANGES_REQUESTED** [review]

File: packages/medusa/src/models/discount-rule-condition.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/medusa/src/models/discount-rule-condition.ts:47]: Good catch! Will change 👍

**Comment by olivermrbl** [comment]
> LGTM - we need passing tests but then merge!

@srindom The tests will come in a later PR, since this is only entity changes and additions. Are you fine with that? I am not merging to`develop`.

**Comment by srindom** [comment]
Got it! Go ahead then :)

**Pull request merged**